flang/lib/Lower/Bridge.cpp

```diff
@@ -1057,7 +1058,38 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     mlir::emitWarning(toLocation(), "ignoring all compiler directives");
   }
 
-  void genFIR(const Fortran::parser::OpenACCConstruct &) { TODO(); }
+  void genFIR(const Fortran::parser::OpenACCConstruct &acc) {
```

Comment:
If you'd rather keep this with the rest of the ACC lowering code in OpenACC.cpp, you could declare this member function here and provide the definition in the OpenACC.cpp file.  (Grouping methods from different classes by an overarching functional concern is done in some areas of LLVM and MLIR, for example.)
It's not clear to me which structure would be more helpful to you.

Reply:
Thanks @clementval for the patch!
Could you please add a test case for nested evaluations as well. 
Maybe we need to address this OpenMPDialect as well ?

Reply:
If we can keep it separate from `Bridge.cpp` that would help avoid noise and bloat in bridge.

Reply:
> If we can keep it separate from `Bridge.cpp` that would help avoid noise and bloat in bridge.

Which should not be readily dismissed given the concerns over upstream patch size.

Reply:
Sure I'll move that code.

Reply:
I moved most of the code back to `OpenACC.cpp`. I guess it is hard to move the last bit since `FirConverter` is declared in this `.cpp` file. Or maybe I miss the point? @schweitzpgi